src/migrations/20230116132109-add-sats-amounts-onchain-payments.ts

```diff
@@ -0,0 +1,91 @@
+const txTypes = ["onchain_payment"]
+
+module.exports = {
+  async up(db) {
+    const collection = db.collection("medici_transactions")
+
+    try {
+      const allTxns = await collection.aggregate([
+        {
+          $match: {
+            type: { $in: txTypes },
+            satsAmount: { $exists: false },
+            currency: "BTC",
+          },
+        },
+        {
+          $group: {
+            _id: "$hash",
+            // For payments, 'currency: "BTC"' filter applies above:
+            // - for USD: max sat debit comes from dealer BTC wallet
+            // - for BTC: max sat debit comes from sender BTC wallet
+            debit: { $max: "$debit" },
+            fee: { $first: "$fee" },
+            usd: { $first: "$usd" },
+            feeUsd: { $first: "$feeUsd" },
+          },
+        },
+      ])
+      for await (const { _id: hash, debit, fee, usd, feeUsd } of allTxns) {
+        if (!hash) continue
+        const satsAmount = Math.round(debit - fee)
+        const centsAmount = Math.round((usd - feeUsd) * 100)
+        const centsFee = Math.round(feeUsd * 100)
+
+        const resultRest = await collection.update(
+          { hash },
+          [
+            {
+              $set: {
+                satsAmount,
+                satsFee: fee,
+                centsAmount,
+                centsFee,
+                displayAmount: centsAmount,
+                displayFee: centsFee,
+                displayCurrency: "USD",
+                satsAmountMigration: true,
```

Comment:
maybe instead of `satsAmountMigration` we can have a field with the current display amount (in that way we can validate or rollback because this is set to empty if we rollback so we will loose the value if that happens)

Reply:
There isn't a property called `displayAmount` for these transactions I think. We use the `usd` property to derive that, which isn't being touched in these migrations

![image](https://user-images.githubusercontent.com/17693119/212744101-2904c4f9-bcd0-49ba-9610-6718954af6b6.png)